### PR TITLE
refactor: Move advanced_cluster common code to cluster 

### PR DESCRIPTION
**Pull request by maastha**
## Description

This PR separates the common code between `mongodbatlas_advanced_cluster` and `mongodbatlas_cluster` resource and data sources. `mongodbatlas_advanced_cluster` is in the process of being migrated to Terraform Plugin Framework, hence any code used by `mongodbatlas_cluster` has been copied over to `mongodbatlas_cluster`'s directory.

These common methods, etc will be removed from `mongodbatlas_advanced_cluster` codebase in future PRs.

- Follow-up PRs (WIP) refactoring advanced_cluster resource and data sources to TF Plugin Framework: 

     - https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1729 

Link to any related issue(s):

## Type of change:

- [ ] Bug fix (non-breaking change which fixes an issue). Please, add the "bug" label to the PR.
- [ ] New feature (non-breaking change which adds functionality). Please, add the "enhancement" label to the PR.
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected). Please, add the "breaking change" label to the PR.
- [ ] This change requires a documentation update
- [ ] Documentation fix/enhancement

## Required Checklist:

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [x] I have read the [contribution guidelines](https://github.com/mongodb/terraform-provider-mongodbatlas/blob/master/CONTRIBUTING.md)
- [x] I have checked that this change does not generate any credentials and that **they are NOT accidentally logged anywhere**.
- [ ] I have added tests that prove my fix is effective or that my feature works per HashiCorp requirements
- [ ] I have added any necessary documentation (if appropriate)
- [x] I have run make fmt and formatted my code
- [ ] If changes include deprecations or removals, I defined an isolated PR with a relevant title as it will be used in the auto-generated changelog.

### CODE REVIEW

**Review by andreaangiolillo — APPROVED** [review]
LGTM

**Comment by maastha** [comment]
cluster test currently failing on master as well:
```
=== NAME  TestAccClusterRSCluster_withDefaultBiConnectorAndAdvancedConfiguration_maintainsBackwardCompatibility
    testing_new.go:80: Error retrieving state, there may be dangling resources: exit status 1
        Failed to marshal state to json: unsupported attribute "bi_connector"
```

File: internal/service/cluster/data_source_cluster.go [hunk 1]
```diff
@@ -25,7 +27,7 @@ func DataSource() *schema.Resource {
 				Type:     schema.TypeString,
 				Required: true,
 			},
-			"advanced_configuration": advancedcluster.ClusterAdvancedConfigurationSchemaComputed(),
+			"advanced_configuration": clusterAdvancedConfigurationSchemaComputed(),
```
Comment (lantoli) [inline internal/service/cluster/data_source_cluster.go:30]: i don't see the functions being removed from advancedcluster package, are we duplicating the functions?

**Comment by lantoli** [comment]
> cluster test currently failing on master as well:
> 
> ```
> === NAME  TestAccClusterRSCluster_withDefaultBiConnectorAndAdvancedConfiguration_maintainsBackwardCompatibility
>     testing_new.go:80: Error retrieving state, there may be dangling resources: exit status 1
>         Failed to marshal state to json: unsupported attribute "bi_connector"
> ```

is it still failing in master and/or in this PR?

**Comment by lantoli** [comment]
if we have common code for cluster and advanced cluster, wouldn't it better to have it in advanced cluster if at the end we'll deprecate cluster?

**Comment by maastha** [comment]
>if we have common code for cluster and advanced cluster, wouldn't it better to have it in advanced cluster if at the end we'll deprecate cluster?

@lantoli common code in advanced_Cluster will be refactored to Plugin Framework, hence it can no longer be used with Cluster, so cluster will need to maintain separate codebase/method implementations until it is deprecated.

File: internal/service/cluster/data_source_cluster.go (hunk 1 above)
Comment (maastha) [reply internal/service/cluster/data_source_cluster.go:30]: yes, advanced cluster will be refactored to framework in follow-up PRs, updated description

**Comment by lantoli** [comment]
> > if we have common code for cluster and advanced cluster, wouldn't it better to have it in advanced cluster if at the end we'll deprecate cluster?
> 
> @lantoli common code in advanced_Cluster will be refactored to Plugin Framework, hence it can no longer be used with Cluster, so cluster will need to maintain separate codebase/method implementations until it is deprecated.

oh, i see, understood, thx!

**Review by AgustinBettati — APPROVED** [review]
LGTM, as @lantoli mentioned would just make sure the failing tests are not related to these changes

**Review by lantoli — APPROVED** [review]

**Comment by marcosuma** [comment]
@maastha unless I missed it, can we change visibility of the `advancedcluster` methods now? I think they are public because were used by cluster, but now no need anymore (not sure if you did it and missed it in the PR).
Similarly to what I said in the other PR, let's wait to release 1.14.0

**Pull request closed without merging**